pandas/core/strings.py

```diff
@@ -323,15 +328,23 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
 
     n : int, default -1 (all)
         Number of replacements to make from start
-    case : boolean, default True
-        If True, case sensitive
-    flags : int, default 0 (no flags)
-        re module flags, e.g. re.IGNORECASE
+    case : boolean, optional
+        - if False, case insensitive
+        - Must be None if `pat` is a compiled regex
```

Comment:
The issue is then you would have to always set to ``None`` if you are using a compiled regex. If we have it as ``True``, then you will either have it ignored or misleading (if it differs from the flags in the regex).

so easy enough to have ``flags=0`` be the default.

@jorisvandenbossche thoughts here?